presto-orc/src/main/java/com/facebook/presto/orc/RowGroup.java

```diff
@@ -28,6 +29,8 @@
 
     public RowGroup(int groupId, long rowOffset, long rowCount, long minAverageRowBytes, InputStreamSources streamSources)
     {
+        checkArgument(rowOffset >= 0, "Invalid row offset %s for group id %s", rowOffset, groupId);
```

Comment:
I guess I understand the rationale but will this help prevent wrapping around? `rowOffset` is calculated by `int * long`, so it is possible to wrap around and still be positive, for example, `3 * Long.MAX_VALUE`.

Reply:
Wrapping around is unlikely, for this to happen the file has to have Long.MAX_VALUE rows or some where close to it. That would be close to 9 exabytes even if the row has a byte value.

Reply:
If we did overflow check when calculating these values, do we still need this non-negative check?

Reply:
Most of the public API's, we do argument checking and this place we lacked one, so I added it. This should have no performance penalty and keeps our API consistent. 

For example, https://github.com/prestodb/presto/blob/master/presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java#L222